src/backend/stream.rs

```diff
@@ -480,7 +525,13 @@ impl<'ctx> PulseStream<'ctx> {
 
             // Set up input stream
             if let Some(stream_params) = input_stream_params {
-                match PulseStream::stream_init(context, stream_params, stream_name) {
+                // Handle loopback audio name
+                let mut actual_stream_name = stream_name;
+                if stream_params.prefs() == StreamPrefs::LOOPBACK {
+                    actual_stream_name = Some(sink_name.as_c_str());
+                }
+
+                match PulseStream::stream_init(context, stream_params, actual_stream_name) {
```

Comment:
Does this work for the input-only stream? 
This seems to work only for the duplex stream and the input device isn't set?

I am not quite sure what the expected behavior for the following cases:
1. the stream is input-only stream
2. the stream is duplex stream but the input device is set by user
3. the stream is duplex stream but the input device is not set

@padenot : What are the expected behaviors for the above cases?

Reply:
For now, it only fetched whatever default sink device it finds. IN THEORY we could implement it finding any sink device, but I don't think there are many use-cases for such behavior and therefore have avoided it.

Case 1:
For input-only streams, (using the error checking code that you suggested that I make) it will return a cubeb_backend::Error::device_unavailable() because there is no sink device you can grab the source from.
Therefore if the library user tried to request a loopback stream and found none, they should be informed about the fact that there are indeed NO loopback streams available.

Case 2:
That should be invalid (for now, given my first and lazy implementation) but in theory we could make it so it can fetch the ``.monitor`` stream of any sink device. Reason being that the code would have to properly parse the input device and activate a sink just so it could grab it's name and use it as a sink. This would require a TON of changes.

Case 3:
It should just grab the default sink device as use it as a source as I've alluded to before.

But yeah, I you want any of these cases to change in behavior, I'd be happy to change. But for now I just made them do whatever my time/intuition could allow for.